### PR TITLE
Fix issue and add feature

### PR DESCRIPTION
**Pull request by parth-simform**
* Fix negative number issue
    * Fixe double line issue
    * Fixe circle is distorting with random range values
    * Add feature for step-up by long-pressing buttons

### CODE REVIEW

File: src/components/RadialSlider/ButtonContent.tsx [hunk 1]
```diff
@@ -11,6 +11,8 @@ const ButtonContent = (props: ButtonProps) => {
       activeOpacity={0.7}
       disabled={disabled}
       onPress={onPress}
+      onLongPress={props.onLongPress}
```
Comment (mukesh-simform) [inline src/components/RadialSlider/ButtonContent.tsx:14]: @parth-simform Please use optional chaining (?.) and also check default values for onLongPress and onPressOut, This shouldn't crash in any case

File: src/components/RadialSlider/RadialSlider.tsx [hunk 2]
```diff
@@ -61,6 +64,22 @@ const RadialSlider = (props: RadialSliderProps & typeof defaultProps) => {
     centerValue,
   } = useRadialSlider(props);
 
+  useEffect(() => {
+    //check max value length
+    const maxLength = ('' + max).length;
```
Comment (mukesh-simform) [inline src/components/RadialSlider/RadialSlider.tsx:69]: @parth-simform Can use please convert that max to string as follows ?
`const maxLength = max?.toString()?.length`

File: src/components/RadialSlider/hooks/useRadialSlider.ts [hunk 3]
```diff
@@ -22,11 +22,31 @@ const useRadialSlider = (props: RadialSliderHookProps) => {
...
-  const isMarkerVariant = variant === Constants.speedoMeterMarker;
+  useEffect(() => {
+    if (isMarkerVariant)
+      if (min < 0) {
+        throw 'Negative number is not allowed';
+      } else if (max < 0) {
+        throw 'Negative number is not allowed';
+      }
+    if (max < min) {
+      throw 'max value should be more than min';
```
Comment (mukesh-simform) [inline src/components/RadialSlider/hooks/useRadialSlider.ts:47]: @parth-simform Update wording as follows and also formate code
`max value should be greater than min`

File: src/components/RadialSlider/ButtonContent.tsx (hunk 1 above)
Comment (parth-simform) [reply src/components/RadialSlider/ButtonContent.tsx]: Resolved

File: src/components/RadialSlider/RadialSlider.tsx (hunk 2 above)
Comment (parth-simform) [reply src/components/RadialSlider/RadialSlider.tsx]: Resolved

File: src/components/RadialSlider/hooks/useRadialSlider.ts (hunk 3 above)
Comment (parth-simform) [reply src/components/RadialSlider/hooks/useRadialSlider.ts]: Resolved

File: src/components/RadialSlider/hooks/useSilderAnimation.ts [hunk 4]
```diff
@@ -39,18 +39,19 @@ const useSilderAnimation = (props: RadialSliderAnimationHookProps) => {
...
   const [value, setValue] = useState(
-    props.value >= min ? props.value || min : min
+    props?.value < min ? min : props?.value > max ? max : props?.value
   );
 
   useEffect(() => {
-    if (props?.value < min) {
-      setValue(min);
-    } else if (props?.value > max) {
+    //if value is more than max then we can set max value
```
[the remark this answers is not in the document]
Comment (parth-simform) [reply src/components/RadialSlider/hooks/useSilderAnimation.ts]: Resolved

**Pull request merged**